### DISCUSSION TITLE
react JsSip x audio

### DISCUSSION THREAD

**Discussion by telmedola** [versatica/JsSIP#880] [General]
Hi!
Please.. anyone can help me!
I tring everthink, but not works.

The call completes, the permission is ok, and etc.. But audio stay mute.
I having try another webphone and my asterisk is fine, works. But when i using my code, don't work.

Any onde can help me please!
Thanks.
[Demo5.tsx.txt](https://github.com/user-attachments/files/17511973/Demo5.tsx.txt)

**Comment by ismaildasci** [comment]
Hey! 👋

I see you’re struggling with audio issues in your WebRTC setup using JsSIP and Asterisk. It looks like you’ve tried setting up remote audio, but it’s staying muted even though everything else seems to work. Here are a few things you could check and try to potentially fix the issue:

### 1. Ensure Proper Media Stream Attachment
It seems like you’re already adding the remote track to the audio element, but there could be issues with permissions or browser policies. Make sure the remote audio is correctly attached with the following code in `discar`:

```javascript
session?.connection.addEventListener('track', (event) => {
  if (event.track.kind === 'audio') {
    const remoteStream = new MediaStream();
    remoteStream.addTrack(event.track);
    if (audioElementRemote.current) {
      audioElementRemote.current.srcObject = remoteStream;
      audioElementRemote.current.play().catch((error) =>
        console.error("Failed to play remote audio:", error)
      );
    }
  }
});
```

### 2. Check Asterisk Configuration
Ensure that Asterisk is configured to allow audio passthrough for WebRTC clients. Confirm that **STUN/TURN servers** are correctly set up to help with NAT traversal if your setup is behind NAT.

### 3. Handle Audio Context Policies
Some browsers (like Chrome) require user interaction to play audio. Try initializing and playing the audio in response to a user action, like clicking a button, to ensure compliance with these policies.

### 4. Verify Media Constraints
Make sure that you are requesting audio permissions correctly in `mediaConstraints`:

```javascript
const options = {
  eventHandlers: eventHandlers,
  mediaConstraints: { audio: true, video: false },
};
```

### 5. Debugging Tips
- Open the browser console and check for errors related to audio playback.
- Use `console.log()` to verify that `track` events are being triggered and that audio tracks are being received.

If these steps don’t resolve the issue, feel free to share more details or any errors you’re seeing. Hope this helps! 😊

**Reply by telmedola** [reply]
Hi, thanks a lot for reply.

I tested my Asterisk with another softphone(paid), and it worked, but here I am unsure about using the stun parameters:

const options : CallOptions = {
          eventHandlers : eventHandlers,
          mediaConstraints : { audio: true, video: false },
          pcConfig: {
            iceServers:[ // optional
              { urls: ['stun:stun.l.google.com:19302', 'stun:stun4.l.google.com:19302'] },
            ]
          },
        };

That's correct? Is there a way to enable JsSip's debug log? Can you help me please?

Thanks

**Comment by telmedola** [comment]
Still can't adjust.
On the first call, when mic and speaker permission is requested, the call happens normally, both sides hear each other. But after the first one is turned off and an attempt is made to make a second one, the sides no longer hear each other. Only after resetting permissions again do both sides hear each other.
